src/libp2p/peer_manager.rs

```diff
@@ -127,22 +127,27 @@ impl PeerManager {
         let mut peers: Vec<_> = peer_lk
             .full_peers
             .iter()
-            .map(|(p, info)| {
-                let cost = if (info.successes + info.failures) > 0 {
-                    // Calculate cost based on fail rate and latency
-                    let fail_rate = f64::from(info.failures) / f64::from(info.successes);
-                    info.average_time.as_secs_f64() + fail_rate * average_time.as_secs_f64()
+            .filter_map(|(p, info)| {
+                // Filter out nodes that are stateless (or far behind)
+                if info.head.epoch() > 0 {
+                    let cost = if (info.successes + info.failures) > 0 {
+                        // Calculate cost based on fail rate and latency
+                        let fail_rate = f64::from(info.failures) / f64::from(info.successes);
```

Comment:
What happens if `info.successes` is `0.0`?

Reply:
Good question, I only moved this piece of code around without inspecting it. I think `if info.success > 0` should be a proper fix